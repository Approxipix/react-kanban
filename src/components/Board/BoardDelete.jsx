import React, { Component } from 'react';
import { connect } from 'react-redux';
import { history } from "../../redux/store";
import { bindActionCreators } from 'redux';
import { deleteBoard } from '../../redux/boardReducer/actions/actions';
import { Text, SubmitButton } from '../BaseComponent';

class BoardDelete extends Component {
  deleteBoard = () => {
    this.props.actions.deleteBoard({
      boardId: this.props.boardId
    });
    history.push('/boards')
  };

  render() {
    return (
      <>
        <Text>
          All actions will be removed from the activity feed and you wonâ€™t be able to re-open the board.
          There is no undo.
        </Text>
        <SubmitButton color={'danger'} onClick={() => this.deleteBoard()}>
          Delete board
        </SubmitButton>
      </>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteBoard: deleteBoard,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(BoardDelete);
