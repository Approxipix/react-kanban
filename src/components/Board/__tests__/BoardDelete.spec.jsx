import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import BoardDelete from '../BoardDelete';

const mockStore = configureStore([]);

describe('<BoardDelete>', () => {
  let store;
  let component;
  let mockToggleSidebar = jest.fn();

  beforeEach(() => {
    const props = { boardId: 'BoardID' };
    store = mockStore({});
    store.dispatch = mockToggleSidebar;

    const wrapper = shallow(<BoardDelete store={store} {...props} />);
    component = wrapper.find('BoardDelete').dive();
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should dispatch an action after click on button", () => {
    const expectedAction = {
      type: 'DELETE_BOARD',
      payload: {
        boardId: 'BoardID'
      }
    };

    component.find('SubmitButton').simulate('click');

    expect(mockToggleSidebar).toHaveBeenCalledTimes(1);
    expect(mockToggleSidebar).toHaveBeenCalledWith(expectedAction);
  });
});

