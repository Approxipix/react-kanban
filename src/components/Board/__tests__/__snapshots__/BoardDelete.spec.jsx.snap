// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardDelete> should render correctly 1`] = `
<Fragment>
  <styled.p>
    All actions will be removed from the activity feed and you wonâ€™t be able to re-open the board. There is no undo.
  </styled.p>
  <SubmitButton
    color="danger"
    onClick={[Function]}
  >
    Delete board
  </SubmitButton>
</Fragment>
`;
